src/client/logging/transports.ts

```diff
@@ -6,10 +6,12 @@ import * as logform from 'logform';
 import * as path from 'path';
 import * as winston from 'winston';
 import * as Transport from 'winston-transport';
+import { IOutputChannel } from '../common/types';
 import { EXTENSION_ROOT_DIR } from '../constants';
 import { LogLevel, resolveLevel } from './levels';
 import { Arguments } from './util';
 
+let pythonOutputChannel: IOutputChannel;
```

Comment:
All global stuff should be in `_global.ts`.  However, keeping a global `pythonOutputChannel` would mean a more complicated approach.  Instead, the simplest solution would be a function in `_global.ts` that fits this specific use case:

```typescript
export function addOutputChannelLogging(config: ChannelConfig) {
    const formatter = getFormatter();
    const transport = getPythonOutputChannelTransport(config, formatter);
    logger.add(transport);
}
```

Where here we have the following:

```suggestion
export type ChannelConfig = {
    channel: IOutputChannel;
};
```

FWIW, the best solution would be to refactor the code in `_global.ts`, `logger.ts`,  and `transports.ts` to allow updating the global logger or even replace it.  (Note that we only need the global logger for the trace decorators.)  However, let's stick with the simpler solution here for now.